src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java

```diff
@@ -692,31 +870,1955 @@ public boolean validatePortRange(Integer portNumber) {
+				streamConversion.setStreamFlipping(streamFlipping.getName());
+				createStream.setStreamConversion(streamConversion);
+				createStream.setStreamFlipping(streamFlipping.getName());
+
+				populateCreateStreamControl(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				break;
+			case SRT_TO_UDP_ADDRESS:
+				streamConversion = createStream.getStreamConversion();
+				streamConversion.setAddress(value);
+				addAdvanceControlProperties(advancedControllableProperties,
+						createText(stats, streamControllingGroup + StreamControllingMetric.SRT_TO_UDP_ADDRESS.getName(), value));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+				break;
+			case SRT_TO_UDP_PORT:
+				streamConversion = createStream.getStreamConversion();
+				String srtToUDPPort = DecoderConstant.EMPTY;
+				try {
+					Integer portIntValue = Integer.parseInt(value);
+					if (portIntValue < DecoderConstant.MIN_PORT) {
+						portIntValue = DecoderConstant.MIN_PORT;
+					}
+					if (portIntValue > DecoderConstant.MAX_PORT) {
+						portIntValue = DecoderConstant.MAX_PORT;
+					}
+					srtToUDPPort = portIntValue.toString();
+				} catch (Exception e) {
+					if (logger.isWarnEnabled()) {
+						logger.warn("Invalid port value", e);
+					}
+				}
+				streamConversion.setUdpPort(srtToUDPPort);
+				createStream.setStreamConversion(streamConversion);
+
+				addAdvanceControlProperties(advancedControllableProperties,
+						createNumeric(stats, streamControllingGroup + StreamControllingMetric.SRT_TO_UDP_PORT.getName(), srtToUDPPort));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+				break;
+			case SRT_TO_UDP_TOS:
+				streamConversion = createStream.getStreamConversion();
+				try {
+					Integer tosIntValue;
+					if (value.startsWith(DecoderConstant.HEX_PREFIX)) {
+						tosIntValue = Integer.parseInt(value.replace(DecoderConstant.HEX_PREFIX, ""), 16);
+					} else {
+						tosIntValue = (int) Float.parseFloat(value);
+					}
+					String tosHexValue = DecoderConstant.HEX_PREFIX + String.format("%02X", 0xFF & tosIntValue);
+					if (tosIntValue < Integer.parseInt(DecoderConstant.MIN_OF_TOS, 16)) {
+						tosHexValue = DecoderConstant.HEX_PREFIX + DecoderConstant.MIN_OF_TOS;
+					}
+					if (tosIntValue > Integer.parseInt(DecoderConstant.MAX_OF_TOS, 16)) {
+						tosHexValue = DecoderConstant.HEX_PREFIX + DecoderConstant.MAX_OF_TOS;
+					}
+					streamConversion.setTos(tosHexValue);
+					createStream.setStreamConversion(streamConversion);
+
+					addAdvanceControlProperties(advancedControllableProperties,
+							createText(stats, streamControllingGroup + StreamControllingMetric.SRT_TO_UDP_TOS.getName(), tosHexValue));
+					populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+					break;
+				} catch (Exception var60) {
+					throw new NumberFormatException("Value of ParameterToS is invalid. TOS must be hex value range to 00-FF");
```

Comment:
Is it possible to add the actual value here as well? So it is clear what caused it exactly. Also, can we tell that the "Exception" will be always caused by this?

Reply:
I fixed it